GArDen/convert/molecule.py

```diff
@@ -0,0 +1,326 @@
+        """Constructor.
+
+        valid 'file_format' strings and what the transformer will expect
+        smiles: list of smiles strings
+        smi: path to .smi file
+        sdf: pat to .sdf file
+        """
+        self.file_format = file_format
+
+    def transform(self, data):
+        """Transform."""
+        try:
+            if self.file_format == 'smiles':
+                graphs = smiles_to_nx(data)
+            elif self.file_format == 'smi':
+                graphs = smi_to_nx(data)
+            elif self.file_format == 'sdf':
+                graphs = sdf_to_nx(data)
+            else:
+                raise Exception('file_format must be smiles smi or sdf')
+            for graph in graphs:
+                yield graph
+
+        except Exception as e:
+            logger.debug('Failed iteration. Reason: %s' % e)
+            logger.debug('Exception', exc_info=True)
+
+
+def set_coordinates(chemlist):
+    for m in chemlist:
+        if m:
+            # updateprops fixes "RuntimeError: Pre-condition Violation"
+            m.UpdatePropertyCache(strict=False)
+            tmp = AllChem.Compute2DCoords(m)
+        else:
+            print '''set coordinates failed..'''
+
+
+def draw(graphs, n_graphs_per_line=5, size=250, title_key=None, titles=None, smiles=False):
+    # we want a list of graphs
+    if isinstance(graphs, nx.Graph):
+        print "give me a list of graphs"
+
+    # make molecule objects
+    chem = [nx_to_rdkit(graph) for graph in graphs]
+    # print chem
+    # calculate coordinates:
+    set_coordinates(chem)
+
+    # take care of the subtitle of each graph
+    if title_key:
+        legend = [g.graph[title_key] for g in graphs]
+    elif titles:
+        legend = titles
+    else:
+        legend = [str(i) for i in range(len(graphs))]
+
+    if smiles == False:
+        # make the image
+        image = Draw.MolsToGridImage(chem, molsPerRow=n_graphs_per_line, subImgSize=(size, size), legends=legend)
+        # display on the spot
+        display(image)
+    else:
+        for m in chem:
+            print Chem.MolToSmiles(m)
+        print '\n'
+
+
+def nx_to_smi(graphs, file):
+    chem = [nx_to_rdkit(graph) for graph in graphs]
+    smis = [Chem.MolToSmiles(m) for m in chem]
+    with open(file, 'w') as f:
+        f.write('\n'.join(smis))
+
+
+def sdf_to_nx(file):
+    # read sdf file
+    suppl = Chem.SDMolSupplier(file)
+    for mol in suppl:
+        yield rdkmol_to_nx(mol)
+
+
+def smi_to_nx(file):
+    # read smi file
+    suppl = Chem.SmilesMolSupplier(file)
+    for mol in suppl:
+        yield rdkmol_to_nx(mol)
+
+
+def rdkmol_to_nx(mol):
+    #  rdkit-mol object to nx.graph
+    graph = nx.Graph()
+    for e in mol.GetAtoms():
+        graph.add_node(e.GetIdx(), label=e.GetSymbol())
+    for b in mol.GetBonds():
+        graph.add_edge(b.GetBeginAtomIdx(), b.GetEndAtomIdx(), label=str(int(b.GetBondTypeAsDouble())))
+    return graph
+
+
+def smiles_to_nx(smileslist):
```

Comment:
smi and smiles are more or less synonyms. I would make the not offer both functions as a public API. EDeN should probably assume always a file object or always a path to make it consistent, the rest is up the user and you should not care about this with more complicated API.